core/embed/rust/src/ui/display/font.rs

```diff
@@ -248,14 +249,33 @@ impl Font {
         (start + end + self.visible_text_height(text)) / 2
     }
 
+    fn get_glyph_data(&self, c: u16) -> *const u8 {
```

Comment:
this is very `unsafe`
but we could make it safe:
1. put a proper lifetime on `Glyph`
2. make a struct `GlyphData` that owns an `Option <RwLockReadGuard<...>>` (the same type that `flash::translations::get()` is returning) and also a `FontInfo` (or `&'static FontInfo` or `&'a FontInfo` or sth :woman_shrugging: )
3. move function `get_glyph` to that struct
4. make private `Font::get_glyph_data` that makes the GlyphData from `self` and `translations::flash::get`
5. make `pub fn Font::with_glyph_data(self, closure)` that will give a new instance of `GlyphData` to the inner closure (and add a comment that we do this to make it somewhat clearer that **GlyphData hold a lock to translations** and that if you store it long-term, you'll break things)
6. rewrite the only user of `get_glyph`, that is `shape::text::Text`, to use `with_glyph_data` in its `draw()` function

Reply:
(and of course make `get_utf8_glyph` would be converted to a method of `Translation`)

Reply:
On it but the intermediate issue is that `Translation` cannot be included for bootloader which makes working with the `RwLock` more complicated.

Reply:
the member of `GlyphData` will be feature-gated then?

Reply:
See here: 02aba6b8d317bef7c4a571a7c7d67064e07b51c2
Handling of translations is feature gated.

Some points to discuss:
- `fn GlypData::get_glyph_data` returns `nullptr` if `get_info` fails
- `pub fn GlyphData::get_glyph` - not sure if the `ensure!` should be there w.r.t. the point above
- `Translations::get_utf8_glyph` is now simplified, perhaps the commentary could reviewed

Reply:
left some suggestions in code. generally: we can stop mucking with pointers which makes some things much nicer now